core/src/subgraph/registrar.rs

```diff
@@ -507,28 +513,48 @@ fn create_subgraph_version(
             version_summaries_after
         }
         SubgraphVersionSwitchingMode::Synced => {
-            // Just set the new version is pending, unless there is no current versions yet
+            // There is a current version. Depending on whether it is synced
+            // or not, make the new version the pending or the current version
             if current_version_id_opt.is_some() {
                 // Previous pending version (if there was one) is no longer pending
+                // Previous current version if it's not fully synced is no
+                // longer the current version
                 let mut version_summaries_after = version_summaries_before
                     .clone()
                     .into_iter()
                     .map(|mut version_summary| {
                         if Some(&version_summary.id) == pending_version_id_opt.as_ref() {
                             version_summary.pending = false;
                         }
+                        if !current_is_synced
+                            && Some(&version_summary.id) == current_version_id_opt.as_ref()
+                        {
+                            // We will make the new version the current version
+                            version_summary.current = false;
+                        }
 
                         version_summary
                     })
                     .collect::<Vec<_>>();
 
-                // Add new version, which will be pending but not current
+                // Determine if this new version should be the pending or
+                // current version. When we add a version to a subgraph that
+                // already has a current version, the new version becomes the
+                // pending version if the current version is synced,
+                // and replaces the current version if the current version is still syncing
+                let (pending, current) = if current_is_synced {
+                    (true, false)
+                } else {
+                    (false, true)
+                };
+
+                // Add new version, setting pending and current appropriately
                 version_summaries_after.push(SubgraphVersionSummary {
                     id: version_entity_id.clone(),
                     subgraph_id: subgraph_entity_id.clone(),
                     deployment_id: manifest.id.clone(),
-                    pending: true,
-                    current: false,
+                    pending: pending,
+                    current: current,
```

Comment:
These can tidy up to just `pending,` and `current,`

Reply:
Done